### PR TITLE
Sl

### PR DESCRIPTION
**Pull request by pltsc18**
Tests to be finalized.

### CODE REVIEW

File: examples/sl/_01_sl_zijlstra.py
Comment (AKuederle) [inline examples/sl/_01_sl_zijlstra.py:50]: We should decide, if we want to work in cm or m and then either change the sensorheight loaded from the data or change the expected input for the algorithm.

File: examples/sl/_01_sl_zijlstra.py
Comment (AKuederle) [inline examples/sl/_01_sl_zijlstra.py:55]: We should add the unit to the variable name so `sensor_height_m` or `sensor_height_cm` depending on the previous comment.

File: examples/sl/_01_sl_zijlstra.py
Comment (AKuederle) [reply examples/sl/_01_sl_zijlstra.py:50]: I think we are using m most of the time right? So then it would make sense to change the SensorHeight? Any opinions?

File: examples/sl/_01_sl_zijlstra.py
Comment (AKuederle) [inline examples/sl/_01_sl_zijlstra.py:45]: I owuld suggest to show it in the example first with the default values (so no madgwick) and then show the option to add Madgwick later in the example

File: mobgap/sl/_stride_length.py
Comment (AKuederle) [inline mobgap/sl/_stride_length.py:165]: Do we need `tuning_cohort`? Is this used for something?

File: mobgap/sl/_stride_length.py
Comment (AKuederle) [inline mobgap/sl/_stride_length.py:205]: We should remove print statements in the final version

File: mobgap/sl/_stride_length.py
Comment (AKuederle) [inline mobgap/sl/_stride_length.py:209]: In Python you don't need/ don't use the square brackets here.

Comment (AKuederle) [inline mobgap/sl/_stride_length.py:209]: Regarding filtering.

When using butter (or any other scipy filter method), always use the `fs=` parameter to pass the sampling rate and provide Wn in Hz.
Further I would always suggest to use a sos filter. Order 4 should be fine with ba, but SOS is always more stable.

Here, I would actually suggest to use the Mobgap filter utilities to keep everything uniform accross the algorithms. I would also allow us to make the filter easily configurable in the future.

File: mobgap/sl/_stride_length.py
Comment (AKuederle) [inline mobgap/sl/_stride_length.py:279]: See filter comments above

File: mobgap/sl/_stride_length.py
Comment (AKuederle) [inline mobgap/sl/_stride_length.py:287]: I think there is a "find_min_max" function implemented for one of the GSD or SL algorithms. We might be able to use this here.

File: mobgap/sl/_stride_length.py
Comment (AKuederle) [inline mobgap/sl/_stride_length.py:164]: I think we can do better with the name here. Maybe `step_length_scaling_factor`?

File: mobgap/sl/demo_sl.py
Comment (AKuederle) [inline mobgap/sl/demo_sl.py:1]: We should remove this before merging

File: mobgap/sl/_stride_length.py
Comment (AKuederle) [inline mobgap/sl/_stride_length.py:249]: Mmh `stpl_list_` might be hard to understand. Maybe we can use a better name. Might be the best to be more vorbose in both cases. So stride_length_per_sec... and step_length_per_sec... to avoid confusion

Comment (AKuederle) [reply mobgap/sl/_stride_length.py:249]: Ah and we might want to have 3 outputs: Raw zjilstra output, interpolated zjilstra, interpolated and converted to strides.

File: mobgap/sl/base.py
Comment (AKuederle) [inline mobgap/sl/base.py:93]: For the base class I would not include stpl_list_, as we don't assume that all potential stride lnegth algorithms can provide it.

**Review by AKuederle — CHANGES_REQUESTED** [review]

File: examples/sl/_01_sl_zijlstra.py
Comment (pltsc18) [reply examples/sl/_01_sl_zijlstra.py:50]: That's fine for me! I am not sure about other cases where Sensor Height is needed, but at least in the Zijlstra formula you need sensor height in meters, so we could directly save the value in meters in the participant's metadata

File: examples/sl/_01_sl_zijlstra.py
Comment (pltsc18) [reply examples/sl/_01_sl_zijlstra.py:55]: I will rename it "sensor_height_m" for now

File: examples/sl/_01_sl_zijlstra.py
Comment (pltsc18) [reply examples/sl/_01_sl_zijlstra.py:45]: Ok!

File: mobgap/sl/_stride_length.py
Comment (pltsc18) [reply mobgap/sl/_stride_length.py:165]: Actually no, I thought it could be useful to hardcode it in case that someone wants to know the cohorts that the values correspond to. but I guess that it can be documented somewhere else (also, it is explicitated in the name of the tuning_coefficient e.g., tuning_coefficient_MS_MS)

File: mobgap/sl/_stride_length.py
Comment (pltsc18) [reply mobgap/sl/_stride_length.py:205]: Done!

File: mobgap/sl/_stride_length.py
Comment (pltsc18) [reply mobgap/sl/_stride_length.py:209]: Thanks!

Comment (pltsc18) [reply mobgap/sl/_stride_length.py:209]: Thanks for the suggestions. I will try and use the MobGap filter. Do you mean an instance to the ButterworthFilter class right?

File: mobgap/sl/_stride_length.py
Comment (pltsc18) [reply mobgap/sl/_stride_length.py:279]: Ok thanks!

File: mobgap/sl/_stride_length.py
Comment (pltsc18) [reply mobgap/sl/_stride_length.py:164]: That's fine for me! I will rename it.

File: mobgap/sl/demo_sl.py
Comment (pltsc18) [reply mobgap/sl/demo_sl.py:1]: Ups, I forgot to remove it, sorry!

File: mobgap/sl/_stride_length.py
Comment (pltsc18) [reply mobgap/sl/_stride_length.py:287]: Ok thanks for the suggestion!

File: mobgap/sl/base.py
Comment (pltsc18) [reply mobgap/sl/base.py:93]: You are right, I removed it from the base class!

File: mobgap/sl/_stride_length.py
Comment (pltsc18) [reply mobgap/sl/_stride_length.py:249]: Seems legit. I renamed the three outputs, now it should be clearer:
- step_length_list_
- step_length_per_sec_list_
- stride_length_per_sec_list_

File: mobgap/sl/_stride_length.py
Comment (pltsc18) [reply mobgap/sl/_stride_length.py:287]: I am sorry, I did not find anything like a find_min_max function... The most similar that I found is a function to find minima between zero crossings, which is called by the detect method of the IcdIonescu class

**Review by AKuederle — APPROVED** [review]

**Pull request merged**
